adapters/mobilefuse/mobilefuse.go

```diff
@@ -0,0 +1,193 @@
+	parsedTemplate, errs := template.New("endpointTemplate").Parse(endpointTemplate)
+
+	if errs != nil {
+		glog.Fatal("Unable parse endpoint template: " + errs.Error())
+		return nil
+	}
+
+	return &MobilefuseAdapter{EndpointTemplate: *parsedTemplate}
+}
+
+func (adapter *MobilefuseAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+
+	adapterRequest, errs := adapter.makeRequest(request)
+
+	if errs == nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	}
+
+	return adapterRequests, errs
+}
+
+func (adapter *MobilefuseAdapter) MakeBids(incomingRequest *openrtb.BidRequest, outgoingRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d.", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d.", response.StatusCode),
+		}}
+	}
+
+	var incomingBidResponse openrtb.BidResponse
+
+	if err := json.Unmarshal(response.Body, &incomingBidResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	outgoingBidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
+
+	for _, seatbid := range incomingBidResponse.SeatBid {
+		for i := range seatbid.Bid {
+			outgoingBidResponse.Bids = append(outgoingBidResponse.Bids, &adapters.TypedBid{
+				Bid:     &seatbid.Bid[i],
+				BidType: adapter.getBidType(seatbid.Bid[i].ImpID, incomingRequest.Imp),
+			})
+		}
+	}
+
+	return outgoingBidResponse, nil
+}
+
+func (adapter *MobilefuseAdapter) makeRequest(bidRequest *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+
+	mobilefuseExtension, errs := adapter.getMobilefuseExtension(bidRequest)
+
+	if errs != nil {
+		return nil, errs
+	}
+
+	endpoint, err := adapter.getEndpoint(mobilefuseExtension)
+
+	if err != nil {
+		return nil, append(errs, err)
+	}
+
+	validImps := adapter.getValidImps(bidRequest, mobilefuseExtension)
+
+	if len(validImps) == 0 {
+		err := fmt.Errorf("No valid imps")
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	mobilefuseBidRequest := *bidRequest
+	mobilefuseBidRequest.Imp = validImps
+	body, err := json.Marshal(mobilefuseBidRequest)
+
+	if err != nil {
+		return nil, append(errs, err)
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     endpoint,
+		Body:    body,
+		Headers: headers,
+	}, errs
+}
+
+func (adapter *MobilefuseAdapter) getMobilefuseExtension(request *openrtb.BidRequest) (*openrtb_ext.ExtImpMobilefuse, []error) {
```

Comment:
Nitpick: Consider renaming to getFirstMobilefuseExtension to make its operation clearer.